libtenzir/builtins/operators/load_tcp.cpp

```diff
@@ -204,19 +207,41 @@ class load_tcp_sink_operator final
   load_tcp_sink_operator() = default;
 
   explicit load_tcp_sink_operator(
-    const connection_manager_actor<Elements>& connection_manager)
-    : connection_manager_{connection_manager} {
+    const connection_manager_actor<Elements>& connection_manager,
+    std::optional<ast::field_path> local_endpoint_field,
+    boost::asio::ip::tcp::socket::endpoint_type local_endpoint)
+    : connection_manager_{connection_manager},
+      local_endpoint_field_{std::move(local_endpoint_field)},
+      local_endpoint_ip_{
+        local_endpoint.address().is_v4()
+          ? ip::v4(as_bytes(local_endpoint.address().to_v4().to_bytes()))
+          : ip::v6(as_bytes(local_endpoint.address().to_v6().to_bytes()))},
+      local_endpoint_port_{local_endpoint.port()} {
   }
 
   auto operator()(generator<Elements> input, operator_control_plane& ctrl) const
     -> generator<std::monostate> {
     auto connection_manager = connection_manager_.lock();
     TENZIR_ASSERT(connection_manager);
+    const auto local_endpoint = ast::constant{
+      record{
+        {"ip", local_endpoint_ip_},
+        {"port", local_endpoint_port_},
```

Comment:
It should be possible to use the `resolver` to do a reverse dns lookup. With that we could add `hostname` and `domain` fields as well. This is something that for example Splunk does by default for their TCP inputs.

Reply:
I added the hostname.